### PR TITLE
Update VMs for July 2022 Patch Tuesday, and to Ubuntu 22.04.

### PR DESCRIPTION
**Pull request by BillyONeal**
* Changes the source from NVidia official for most CUDA things to Ubuntu's built in bits. Note that Ubuntu has no nccl so we're still getting that from them directly.
* Updates PowerShell to 7.2.5.

### CODE REVIEW

File: scripts/azure-pipelines/linux/provision-image.sh [hunk 1]
```diff
@@ -56,22 +63,28 @@ APT_PACKAGES="$APT_PACKAGES libxdamage-dev"
 # Additionally required by gtk3 and at-spi2-atk
 APT_PACKAGES="$APT_PACKAGES libdbus-1-dev"
```
Comment (JackBoosY) [inline scripts/azure-pipelines/linux/provision-image.sh:64]:
```suggestion
APT_PACKAGES="$APT_PACKAGES libdbus-1-dev libxtst-dev"
```

Comment (BillyONeal) [reply scripts/azure-pipelines/linux/provision-image.sh:64]: It's kinda too late. When I make this PR the image with the new configuration is already minted. :(

**Comment by BillyONeal** [comment]
Full build was totally green: https://dev.azure.com/vcpkg/public/_build/results?buildId=75152&view=results

(osx still running but a VM update doesn't affect osx)

Will merge as soon as PR run is green too.

File: scripts/azure-pipelines/linux/provision-image.sh (hunk 1 above)
Comment (dg0yt) [reply scripts/azure-pipelines/linux/provision-image.sh:64]: IIUC it is included three lines later?

Comment (BillyONeal) [reply scripts/azure-pipelines/linux/provision-image.sh:64]: > IIUC it is included three lines later?

I don't see it

image

Comment (talregev) [reply scripts/azure-pipelines/linux/provision-image.sh:64]: It should be later in the script.

Comment (BillyONeal) [reply scripts/azure-pipelines/linux/provision-image.sh:64]: Maybe it's being dragged in as a transitive dependency.

Comment (talregev) [reply scripts/azure-pipelines/linux/provision-image.sh:64]: Can you show screen shots later on the script? It should near the line of `libdbus-dev`

Comment (talregev) [reply scripts/azure-pipelines/linux/provision-image.sh:64]: It looks like the old version of the scripts. The changes from this PR didn't take the effects. Can you fix that?

Comment (BillyONeal) [reply scripts/azure-pipelines/linux/provision-image.sh:64]: > It looks like the old version of the script. The changes from this PR don't show up. Can you fix that?

Derp :)

Comment (talregev) [reply scripts/azure-pipelines/linux/provision-image.sh:64]: > > It looks like the old version of the script. The changes from this PR don't show up. Can you fix that?
> 
> Derp :)


How can we fix it?

Comment (BillyONeal) [reply scripts/azure-pipelines/linux/provision-image.sh:64]: Whack Bill over the head

**Pull request merged**